pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -480,6 +480,17 @@ func desiredLoadBalancerService(ci *operatorv1.IngressController, deploymentRef
 			if !isInternal {
 				service.Annotations[alibabaCloudLBAddressTypeAnnotation] = alibabaCloudLBAddressTypeInternet
 			}
+		case configv1.OpenStackPlatformType:
+			// Set a floating IP only if the load balancer scope is external.
+			if !isInternal {
+				if lb != nil && lb.ProviderParameters != nil &&
+					lb.ProviderParameters.Type == operatorv1.OpenStackLoadBalancerProvider &&
+					lb.ProviderParameters.OpenStack != nil {
+					// We know that LoadBalancerIP is deprecated but in this current version of Service, this won't be removed.
+					// It'll probably be removed in Service v2 which is not going to happen in the near future.
+					service.Spec.LoadBalancerIP = lb.ProviderParameters.OpenStack.FloatingIP
```

Comment:
This logic needs to now use only the IngressController spec to determine the desired FloatingIP, not status (which is the `lb` variable here).